app/controllers/api/v1x0/requests_controller.rb

```diff
@@ -1,64 +1,34 @@
 module Api
   module V1x0
     class RequestsController < ApplicationController
-      include Api::V1x0::Mixins::IndexMixin
-      include Api::V1x0::Mixins::RBACMixin
-
-      PERSONA_ADMIN     = 'approval/admin'.freeze
-      PERSONA_APPROVER  = 'approval/approver'.freeze
-      PERSONA_REQUESTER = 'approval/requester'.freeze
-
-      before_action :read_access_check, :only => %i[show]
-      before_action :create_access_check, :only => %i[create]
+      include Mixins::IndexMixin
 
       def create
+        authorize Request
+
         req = RequestCreateService.new.create(params_for_create)
         json_response(req, :created)
       end
 
       def show
         req = Request.find(params.require(:id))
+        authorize req
+
         request.path.end_with?("/content") ? json_response(req.content) : json_response(req)
       end
 
       def index
-        collection(index_scope(requests_prefilter))
+        collection(policy_scope(requests_prefilter))
       end
 
       private
 
       def requests_prefilter
-        return nil unless params[:request_id]
+        return Request unless params[:request_id]
 
-        resource_check('read', params[:request_id], Request) # NotAuthorizedError if current user cannot access parent request
+        authorize Request.find(params.require(:request_id))
         Request.find(params[:request_id]).children
       end
-
-      def rbac_scope(relation)
```

Comment:
@hsong-rh I think we might still need the rbac scope which can be called for index calls. This rbac_sope should also be called from the graphql overlay.

Reply:
No, the logic of `rbac_scope` is moved to `resolve` method, called by `policy_scope`.